rapid7_insight_agent/icon_rapid7_insight_agent/util/graphql_api/api_connection.py

```diff
@@ -0,0 +1,345 @@
+        agents.extend(self._get_agents_from_result_object(results_object))
+        self.logger.info(f"Initial agents received.")
+
+        agent = self._find_agent_in_agents(agents, agent_input, agent_type)
+        if agent:
+            return agent
+
+        # See if we have more pages of data, if so get next page and append until we reach the end
+        self.logger.info(f"Extra pages of agents: {has_next_page}")
+        while has_next_page:
+            self.logger.info(f"Getting next page of agents.")
+            has_next_page, results_object, next_agents = self._get_next_page_of_agents(results_object)
+            agent = self._find_agent_in_agents(next_agents, agent_input, agent_type)
+            if agent:
+                return agent
+
+        raise APIException(cause=f"Could not find agent matching {agent_input} of type {agent_type}.",
+                           assistance=f"Check the agent input value and try again.",
+                           data="NA")
+
+    def _get_next_page_of_agents(self,
+                                 results_object: dict
+                                 ) -> (bool, dict, list):
+        """
+        In the case of multiple pages of returned agents, this will go through each page and append
+        those agents to the agents list
+
+        :param results_object: dict
+        :return: tuple (boolean, dict (results), list (agents))
+        """
+        self.logger.info(f"Getting next page of agents.")
+        next_cursor = results_object.get("data").get("organization").get("assets").get("pageInfo").get("endCursor")
+        payload = {
+            "query": "query( $orgId:String! $nextCursor:String! ) { organization(id: $orgId) { assets( first: 10000 after: $nextCursor ) { pageInfo { hasNextPage endCursor } edges { node { host { id vendor version description hostNames { name } primaryAddress { ip mac } uniqueIdentity { source id } attributes { key value } } id agent { id } } } } } }",
+            "variables": {
+                "orgId": self.org_key,
+                "nextCursor": next_cursor
+            }
+        }
+        results_object = self._post_payload(payload)
+
+        has_next_page = results_object.get("data").get("organization").get("assets").get("pageInfo").get(
+            "hasNextPage")
+
+        next_agents = self._get_agents_from_result_object(results_object)
+
+        return has_next_page, results_object, next_agents
+
+    def _find_agent_in_agents(self,
+                              agents: list,
+                              agent_input: str,
+                              agent_type: str
+                              ) -> Any:
+        """
+        Given a list of agent objects, find the agent that matches our input.
+
+        If no agent is found this will return None
+
+        :param agents: list (agents)
+        :param agent_input: String (Input value to look for)
```

Comment:
```suggestion
        :param agent_input: String (Input value to look for)
```
I thought agent_input was for mac/ip/hostname, based off of the other functions...?